docs/howto/logging.txt

```diff
@@ -0,0 +1,339 @@
+name to determine the loggers for which it will process records).
+
+It will forward messages of levels ``DEBUG`` and higher to the handler named
+``file``.
+
+Note that a logger can forward messages to multiple handlers, so
+the relation between loggers and handlers is many-to-many.
+
+If you execute::
+
+    logger.debug('Attempting to connect to API')
+
+in your code, you will find that message in the file ``general.log`` in the
+root of the project.
+
+Configure a formatter
+~~~~~~~~~~~~~~~~~~~~~
+
+By default, the final log output contains the message part of each
+:class:`log record <logging.LogRecord>`. Use a formatter if you want to
+include additional data. First define your formatters, for example:
+
+..  code-block:: python
+    :emphasize-lines: 3-12
+
+    LOGGING = {
+        [...]
+        'formatters': {
+            'verbose': {
+                'format': '{name} {levelname} {asctime} {module} {process:d} {thread:d} {message}',
+                'style': '{',
+            },
+            'simple': {
+                'format': '{levelname} {message}',
+                'style': '{',
+            },
+        },
+    }
+
+(The ``style`` keyword allows you to specify ``{`` for :meth:`str.format` or
+``$`` for :class:`string.Template` formatting; the default is ``$``.)
+
+See :ref:`logrecord-attributes` for the :class:`~logging.LogRecord` attributes
+you can include.
+
+To apply a formatter to a handler, configure the handler, for example:
+
+..  code-block:: python
+    :emphasize-lines: 5
+
+    'handlers': {
+        'file': {
+            'class': 'logging.FileHandler',
+            'filename': 'general.log',
+            'formatter': 'verbose',
+        },
+    },
+
+.. _naming-loggers:
+
+Use logger namespacing
+~~~~~~~~~~~~~~~~~~~~~~
+
+The unnamed logging configuration ``''`` captures logs from any Python
+application. A named logging configuration will capture logs only from loggers
+with matching names.
+
+A mapping named ``django`` would only process records from logger instances in
+the ``django`` namespace. A mapping named ``django.contrib`` would be even more
+restrictive.
+
+The namespace of a logger instance is defined using
+:py:func:`~logging.getLogger`. For example in ``views.py`` of ``my_app``::
+
+    logger = logging.getLogger(__name__)
+
+will create a logger in the ``my_app.views`` namespace. ``__name__`` allows you
+to organize log messages according to their provenance within your project's
+applications automatically. It also ensures that you will not experience name
+collisions.
+
+A logger mapping named ``my_app`` will capture records from this logger, but
+ignore those from (say) the ``django`` namespace:
+
+..  code-block:: python
+    :emphasize-lines: 4
+
+    LOGGING = {
+        [...]
+        'loggers': {
+            'my_app': {
+                ...
+            },
+        },
+    }
+
+A logger mapping named ``my_app.views`` will only capture records from that
+module:
```

Comment:
It may not matter in the scheme of things, but it's not _really_ that 'module' because as is suggested immediately hereafter in the document, you can explicitly do `logger = logging.getLogger('my_app.views'); logger.warning('...')` from elsewhere, non? 
It's not _common_, though.

Reply:
Thanks for catching this one, I have been trying to move away from the language of modules to talk instead about the loggers' namespaces.